### PR TITLE
4.x orm deprecations

### PR DESCRIPTION
**Pull request by ADmad**
I have left the deprecated methods in `TableRegistry`. Just wanted to confirm we want to get rid of them in 4.x.

Refs #11934

### CODE REVIEW

File: src/ORM/Query.php [hunk 1]
```diff
@@ -455,13 +429,8 @@ public function contain($associations = null, $override = false)
...
 
-        if ($associations === null) {
-            deprecationWarning(
-                'Using Query::contain() as getter is deprecated. ' .
-                'Use getContain() instead.'
-            );
-
-            return $loader->getContain();
+        if ($associations === null && $override === false) {
+            throw new InvalidArgumentException('$associations can be null only $override is true.');
```
Comment (saeideng) [inline src/ORM/Query.php]: but `associations === null ` deprecated 
and we have `clearContain()` method for clearing contain 
`->contain( null, true)` is pointless IMO

Comment (ADmad) [reply src/ORM/Query.php]: I guess we can remove this check all together and one can just use `contain([], true)` to clear/reset.

File: src/ORM/Locator/LocatorInterface.php [hunk 2]
```diff
@@ -24,17 +24,6 @@
...
-
-    /**
-     * Stores a list of options to be used when instantiating an object
-     * with a matching alias.
-     *
-     * @param string|null $alias Name of the alias
-     * @param array|null $options list of options for the alias
-     * @return array The config data.
-     */
-    public function config($alias = null, $options = null);
```
Comment (markstory) [inline src/ORM/Locator/LocatorInterface.php]: Is this class on your list of classes that need `@method` turned into real methods?

Comment (ADmad) [reply src/ORM/Locator/LocatorInterface.php]: Oui monsieur https://github.com/cakephp/cakephp/issues/12094

**Comment by ADmad** [comment]
@markstory @lorenzo  Should I remove deprecated methods from `TableRegistry` too?

**Pull request merged**